File: my_ws/src/my_cpp_pkg/src/hw_status_publisher.cpp

#include <iostream>
#include "/opt/ros/humble/include/rclcpp/rclcpp/rclcpp.hpp"
#include "/opt/ros/humble/include/std_msgs/std_msgs/msg/string.hpp"
// #include "my_robot_interfaces/msg.HardwareStatus.hpp"
#include "/home/prajesh/my_ws/install/my_robot_interfaces/include/my_robot_interfaces/my_robot_interfaces/msg/hardware_status.hpp"
#include <string>

using namespace std;

class MyNode: public rclcpp::Node
{
    private:
        rclcpp::TimerBase::SharedPtr timer_;
        int count;
        // rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub;
        rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr pub;
        void pub_fun()
        {
            count ++;
            auto msg = my_robot_interfaces::msg::HardwareStatus();
            // msg.data = std::string("robot cpp news ") + std::to_string(count);
            msg.temperature = 57;
            msg.motor_status = true;
            msg.debug_message = "nothign to debug cpp";
            pub->publish(msg);
        }
        

    public:
        MyNode(): Node("Robot_news_station")
        {
            RCLCPP_INFO(this->get_logger(), "Robot cpp news station active-> ");
            pub = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("robot_news", 10);
            timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&MyNode::pub_fun, this));
        }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_ws/src/my_py_pkg/my_py_pkg/hw_status_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String
from my_robot_interfaces.msg import HardwareStatus

class my_node(Node):
    
    def __init__(self):
        self.count = 0
        super().__init__("HardwareStatus_pub")
        self.publisher = self.create_publisher(HardwareStatus, "robot_news", 10)
        self.timer = self.create_timer(0.5, self.pub_fun)
        self.get_logger().info("HardwareStatus_pub active->")
    
    def pub_fun(self):
        self.count += 1
        msg = HardwareStatus()
        msg.temperature = 45
        msg.motor_status = True
        msg.debug_message = "Nothing to debug"
        # msg.data = "latest news " + str(self.count)
        self.publisher.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = my_node()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()